Tests/MapboxMapsTests/Foundation/Extensions/Core/DownloadStatusTests.swift

```diff
@@ -0,0 +1,206 @@
+
+        if httpResult.isValue() {
+            XCTAssertEqual(httpResult.value, httpResponseData, "The httpResult should be equal to the input data value.")
+        } else {
+            XCTFail("httpResult should be a value.")
+        }
+    }
+
+    func testConvenienceInitFailure() throws {
+        let status = DownloadStatus(downloadId: 1,
+                                            state: .finished, error: downloadError, totalBytes: 1234,
+                                            receivedBytes: 1234, transferredBytes: 1234,
+                                            downloadOptions: downloadOptions,
+                                            httpResult: .failure(httpRequestError))
+
+        let httpResult = status.__httpResult as! Expected<HttpResponseData, HttpRequestError>
+        let error = try XCTUnwrap(status.error, "Expected a download error")
+        if httpResult.isError() {
+            XCTAssertEqual(httpResult.error, httpRequestError, "The initial error should equal the error for __httpResult.error")
+            XCTAssertEqual(error, downloadError, "The error for the DownloadStatus should equal the initial download error.")
+        } else {
+            XCTFail("The __httpResult should be an error.")
+        }
+    }
+
+    func testNilHttpResult() {
+        let status = DownloadStatus(downloadId: 1,
+                                    state: .finished, error: .none, totalBytes: 1234,
+                                            receivedBytes: 1234, transferredBytes: 1234,
+                                            downloadOptions: downloadOptions,
+                                            httpResult: .success(httpResponseData))
+        status.__httpResult = nil
+        XCTAssertNil(status.__httpResult)
+        XCTAssertNil(status.httpResult)
+    }
+
+    func testValueHttpResult() throws {
+        let httpResult = Expected<HttpResponseData, AnyObject>(value: httpResponseData)
+        XCTAssertEqual(httpResult.value, httpResponseData)
+        XCTAssertNil(httpResult.error, "The error should be nil.")
+
+        let httpResultWrapped: Result<HttpResponseData, Error> = .success(httpResult.value)
+        let result = try XCTUnwrap(httpResultWrapped.get())
+        XCTAssertEqual(result, httpResponseData, "The two HttpResponseData va;ues should be equal.")
+    }
+
+    func testErrorHttpResult() throws {
+        let httpResult = Expected<AnyObject, HttpRequestError>(error: httpRequestError)
+        XCTAssertEqual(httpResult.error, httpRequestError)
+        XCTAssertNil(httpResult.value, "The value should be nil.")
+
+        let httpErrorWrapped: Result<HttpResponseData, HttpRequestError> = .failure(httpRequestError)
+        guard case .failure(let error) = httpErrorWrapped else {
+            XCTFail("Not a failure")
+            return
+        }
+
+        XCTAssertEqual(error, httpRequestError, "The two errors should be equal.")
+    }
```

Comment:
These tests should be covering the behavior of the DownloadStatus.httpResult property. Sorry that wasn't clear in. my earlier comment.